From 1b06ce9dd5363dad2bcc3d2dcdc17279f0cabe43 Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@squashfs.org.uk>
Date: Wed, 10 Sep 2014 22:27:48 +0100
Subject: [PATCH 095/120] action: chmod: fix mode setting if octal value
 specified for mode

The set code incorrectly did (st_mode & ~S_IFMT) | mode, which is the
exact inverse of what it should have been, i.e.  (st_mode & S_IFMT) | mode

Signed-off-by: Phillip Lougher <phillip@squashfs.org.uk>
---
 squashfs-tools/action.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/squashfs-tools/action.c b/squashfs-tools/action.c
index 4e64a33..b70cd26 100644
--- a/squashfs-tools/action.c
+++ b/squashfs-tools/action.c
@@ -1432,7 +1432,7 @@ static int mode_execute(struct mode_data *mode_data, int st_mode)
 
 		switch(mode_data->operation) {
 		case ACTION_MODE_OCT:
-			st_mode = (st_mode & ~S_IFMT) | mode;
+			st_mode = (st_mode & S_IFMT) | mode;
 			break;
 		case ACTION_MODE_SET:
 			st_mode = (st_mode & ~mode_data->mask) | mode;
-- 
2.8.3

